designs/0031-deprecation-schedule.md

```diff
@@ -0,0 +1,254 @@
+- Feature Name: deprecation_schedule
+- Start Date: 2021-09-21
+- RFC PR: (leave this empty)
+- Stan Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Introduce a process by which minor features can be removed from the Stan
+language following a 1 year (3 minor release) period of deprecation. 'Minor
```

Comment:
Is there a reason this only discusses the language and not the rest of Stan?  Could we make the policy universal across repos?  The only concern I have is that the math lib has never respected backward compatibility release over release, at least at the detailed level.  (I think what we need to do is clearly indicate a non-changing public API for the math lib that won't change.)

Reply:
I focused it on the language mainly because I think that is where the primary examples of where we currently have issues with this arise. I think that pystan, cmdstanr, etc will all follow the conventions of their own ecosystems, and recently the Math repo I think formally stated they're not making backwards compatibility guarantees. 

I'm happy to extend it to other parts of the ecosystem, but I'd need help imagining what that looks like for things outside the language.

Reply:
If you want to keep it to the language, I think the title should change from `deprecation_schedule` to make that clear.  I don't know where the convention came from of naming these like file names without spaces.